examples/scenes/src/cosmic_text_scene.rs

```diff
@@ -0,0 +1,450 @@
+                            let c_start = glyph.start + i;
+                            let c_end = glyph.start + i + c.len();
+                            if (start.line != line_i || c_end > start.index)
+                                && (end.line != line_i || c_start < end.index)
+                            {
+                                range_opt = match range_opt.take() {
+                                    Some((min, max)) => Some((
+                                        cmp::min(min, c_x as i32),
+                                        cmp::max(max, (c_x + c_w) as i32),
+                                    )),
+                                    None => Some((c_x as i32, (c_x + c_w) as i32)),
+                                };
+                            } else if let Some((min, max)) = range_opt.take() {
+                                buffer_line.glyph_highlights.push(Rect::from_origin_size(
+                                    Point::new(min as f64, line_top),
+                                    Size::new(cmp::max(0, max - min) as f64, line_height),
+                                ));
+                            }
+                            c_x += c_w;
+                        }
+                    }
+
+                    if layout_run.glyphs.is_empty() && end.line > line_i {
+                        // Highlight all internal empty lines
+                        range_opt = Some((0, buffer.size().0.unwrap_or(0.0) as i32));
+                    }
+
+                    if let Some((mut min, mut max)) = range_opt.take() {
+                        if end.line > line_i {
+                            // Draw to end of line
+                            if layout_run.rtl {
+                                min = 0;
+                            } else {
+                                max = buffer.size().0.unwrap_or(0.0) as i32;
+                            }
+                        }
+                        buffer_line.glyph_highlights.push(Rect::from_origin_size(
+                            Point::new(min as f64, line_top),
+                            Size::new(cmp::max(0, max - min) as f64, line_height),
+                        ));
+                    }
+                }
+            }
+
+            // Cursor
+            if let Some((x, y)) = cursor_position(&editor_info.cursor, &layout_run) {
+                buffer_line.cursor = Some(Rect::from_origin_size(
+                    Point::new(x as f64, y as f64),
+                    Size::new(1.0, line_height),
+                ));
+            }
+        }
+
+        for glyph in layout_run.glyphs {
+            let mut glyph_color = match glyph.color_opt {
+                Some(color) => Color::from_rgba8(color.r(), color.g(), color.b(), color.a()),
+                None => text_color,
+            };
+
+            if let Some(editor_info) = &editor_info {
+                if text_color.components != editor_info.selected_text_color.components {
```

Comment:
This isn't the right way to compare, but I understand that it is the easier for now.

We're addressing this in https://github.com/linebender/color/pull/76 but it'll take some time, so this can get fixed after something else lands and gets propagated here.

Reply:
If you rebase on top of the current `main` as of a couple of minutes ago, the color now impl's `PartialEq`, so you can just `if text_color != ...` instead of looking at the components.